x/exp/batch/batch.go

```diff
@@ -220,10 +220,11 @@ func doBatch(ctx context.Context, be *batchEvaler) error {
 	// then loop the current variable
 	loopEnv := *be.env
 	u := be.Variables[0]
-	_, chPrincipal := cloneSub(be.env.Principal, u.Key, nil)
-	_, chAction := cloneSub(be.env.Action, u.Key, nil)
-	_, chResource := cloneSub(be.env.Resource, u.Key, nil)
-	_, chContext := cloneSub(be.env.Context, u.Key, nil)
+	dummyVal := types.True
+	_, chPrincipal := cloneSub(be.env.Principal, u.Key, dummyVal)
```

Comment:
@philhassey, I think the use of this dummy value here is okay. Let me know if you have any other more clever ideas how to get this to work. The problem is that `Set` tries to call `.hash()` on `nil` here, which obviously doesn't work.

Reply:
That's fine.